tests/std/tests/P0753R2_manipulators_for_cpp_synchronized_buffered_ostream/test.cpp

```diff
@@ -77,7 +78,8 @@ void test_osyncstream_manipulators(string_buffer<typename Ty::char_type, ThrowOn
 
     assert(addressof(flush_emit(os)) == addressof(os));
     if constexpr (is_base_of_v<basic_osyncstream<char_type, traits_type, Alloc>, Ty>) {
-        assert(os.rdstate() == ios::goodbit);
+        auto bit = (buf && buffer_can_sync) ? ios::goodbit : ios::badbit;
+        assert(os.rdstate() == bit);

```

Comment:
I would prefer a better name and `const` (Yeah I am being super nitpicky)
```suggestion
        const auto state = (buf && buffer_can_sync) ? ios::goodbit : ios::badbit;
        assert(os.rdstate() == state);
```